jOOQ/src/main/java/org/jooq/impl/AbstractName.java

```diff
@@ -86,7 +86,13 @@ public final DerivedColumnListImpl fields(String... fieldNames) {
         return new DerivedColumnListImpl(last(), fieldNames);
     }
 
+    @Override
+    public final DerivedColumnListImpl fields(Name... fieldNames) {
+        if (getName().length != 1)
+            throw new IllegalStateException("Cannot create a DerivedColumnList from a qualified name : " + Arrays.asList(getName()));
 
+        return new DerivedColumnListImpl(last(), Tools.fieldNamesToStrings(fieldNames));
```

Comment:
It's ok. If it's already fixed (whole DerivedColumnListImpl to Name and Name[] fields) then there is nothing to repair and this PR is not needed.